lib/application.js

```diff
@@ -42,9 +41,14 @@ module.exports = class Application extends Emitter {
     this.middleware = [];
     this.subdomainOffset = 2;
     this.env = process.env.NODE_ENV || 'development';
-    this.context = Object.create(context);
-    this.request = Object.create(request);
-    this.response = Object.create(response);
+
+    this.Context = class AppContext extends Context {};
+    this.Request = class AppRequest extends Request {};
+    this.Response = class AppResponse extends Response {};
+
+    this.context = this.Context.prototype;
+    this.request = this.Request.prototype;
+    this.response = this.Response.prototype;
   }
```

Comment:
most of it looks good to me, but I do think we should either ditch the prototypes or ditch the classes there, both is a little confusing. In general the use-case was to extend those for your app as you need, so the protos make more sense here to me at least